### DISCUSSION TITLE
Using power user without opening a workspace - "Detecting dbt" forever

### DISCUSSION THREAD

**Discussion by thomasaarholt** [AltimateAI/vscode-dbt-power-user#374] [Q&A] [answered]
I can successfully use dbt power user if I use `code dbt-models` (our dbt repo), and navigate through sql files within that repo with the workspace open. 

Sometimes I just want to open a specific sql file from the terminal without first opening a workspace, e.g. `code dbt-models/path/to/file.sql`. 

I've set the following two options in order for power user to find the python environment that dbt is installed in, and to find the dbt profiles.

```json
    "dbt.dbtPythonPathOverride": "/Users/thomas/oda/dbt-models/.direnv/python-3.9.9/bin/python",
    "dbt.profilesDirOverride": "/Users/thomas/oda/dbt-models",
```
Here I have copied out a file onto my desktop and opened it there. The Python environment is successfully sourced in the terminal, and `dbt --version` runs. But the status bar shows "Detecting dbt forever".

<img width="863" alt="image" src="https://user-images.githubusercontent.com/2721423/225263490-e5942ca7-700d-4862-82f3-11d2def79d9a.png">

In VSCode, `Output -> Log - dbt` shows no output at all. 

With the above settings, if I `code dbt-models` (opening the repo as a workspace), everything works.
Do you have any advice?

We're currently locked in on dbt v1.3.2 and snowflake plugin v1.3.0.

**Comment by mdesmet** [comment, marked as answer]
I think we require a workspace being selected. I also doubt multiple workspaces works at this time.
